engine/src/witness/sol/egress_witnessing.rs

```diff
@@ -0,0 +1,114 @@
+use crate::sol::{
+	commitment_config::CommitmentConfig,
+	retry_rpc::{SolRetryRpcApi, SolRetryRpcClient},
+	rpc_client_api::{
+		RpcTransactionConfig, TransactionConfirmationStatus, TransactionStatus,
+		UiTransactionEncoding,
+	},
+};
+use anyhow::Result;
+use cf_chains::sol::{SolSignature, LAMPORTS_PER_SIGNATURE};
+use itertools::Itertools;
+
+pub async fn get_finalized_fee(
+	sol_client: &SolRetryRpcClient,
+	signature: SolSignature,
+) -> Result<u64> {
+	match sol_client
+		.get_signature_statuses(&[signature], false)
+		.await
+		.value
+		.iter()
+		.exactly_one()
+		.expect("We queried for exactly one signature.")
+	{
+		Some(TransactionStatus {
+			confirmation_status: Some(TransactionConfirmationStatus::Finalized),
+			..
+		}) => {
+			let transaction_meta = sol_client
+				.get_transaction(
+					&signature,
+					RpcTransactionConfig {
+						encoding: Some(UiTransactionEncoding::Json),
+						// Using finalized there could be a race condition where this
+						// doesn't get the tx. But "Processed" is timing out so we better
+						// retry with finalized.
+						commitment: Some(CommitmentConfig::finalized()),
+						// Getting also type 0 even if we don't use them atm
```

Comment:
this comment doesn't make sense to me

Reply:
Me neither - is copy-pasted from the previous impl. @albert-llimos ?

Reply:
no, this comment is correct.
There are legacy transactions and type 0 transactions. Type 0 being one version after legacy, it's just the naming that is confusing.
By default legacy transactions are queried, which are also the transactions we craft. However, we are querying max to type 0 just to query any possible transaction regardless of the type. As explained in the comment we could use None to just query legacy, which would be fine, but I thought I'd just extend it to all types just in case.

Reply:
Yeah not doubting it's correctness... just the phrasing before I can comment on the correctness

Reply:
gotcha. Added a suggestion.

Reply:
I updated the comment.